src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,158 @@
-//       onChange({ ITFStatus });
-//       return ITFStatus === 'active';
-//     },
-//     ({ errors }) => {
-//       const errorMessage = 'Network request failed';
-//       onChange({
-//         ITFStatus,
-//         errorMessage
-//       });
-//       Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
-//     }
-//   );
-// };
-// }
-/* eslint-enable */
+export function resetPrestartState() {
+  return {
+    type: PRESTART_STATE_RESET
+  };
+}
+
+export function resetPrestartDisplay() {
+  return {
+    type: PRESTART_DISPLAY_RESET
+  };
+}
+
+// Returns most recent timestamp from a list of timestamps
+export const getLatestTimestamp = (timestamps) => timestamps.sort((a, b) => new Date(b) - new Date(a))[0];
+export const isNotEmptyObject = (object) => !(Object.keys(object).length === 0 && object.constructor === Object);
+export const getITFsByStatus = (itfs, status) => itfs.filter(itf => itf.status === status);
+
+export const handleCheckSuccess = (data, dispatch) => {
+  // If no existing ITF(s) are found, we will receive an empty {} response
+  const itfList = isNotEmptyObject(data) && data.attributes.intentToFile;
+
+  // If the user does not have any existing ITFs, create one
+  if (!itfList) {
+    return true;
+  }
+
+  // Check for an active ITF, the user should only have one
+  const activeITF = getITFsByStatus(itfList, ITF_STATUSES.active)[0];
+  const expiredITFs = getITFsByStatus(itfList, ITF_STATUSES.expired);
+  // If the user has an active ITF, set currentExpirationDate
+  if (activeITF) {
+    dispatch(setPrestartData({ currentExpirationDate: activeITF.expirationDate }));
+    dispatch(setPrestartStatus(PRESTART_STATUSES.succeeded));
+    return false;
+  }
+  // If the user doesn't have any active ITFs
+  // Check for expired ITFs, and if found, use the latest to set previousExpirationDate
+  // and create a new ITF
+  if (expiredITFs.length > 0) {
+    const latestExpiredITFExpirationDate = getLatestTimestamp(expiredITFs.map(itf => itf.expirationDate));
+    dispatch(setPrestartData({ previousExpirationDate: latestExpiredITFExpirationDate }));
+    return true;
+  }
+
+  // If the user does not have any expired or active ITFs, create one
+  return true;
+};
 
+export const handleCheckFailure = (dispatch) => {
+  dispatch(setPrestartStatus(PRESTART_STATUSES.failed));
+  return false;
+};
+
+export function checkITFRequest(dispatch) {
+  dispatch(setPrestartData({ verificationType: PRESTART_VERIFICATION_TYPES.retrieve }));
+
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => handleCheckSuccess(data, dispatch),
+    () => handleCheckFailure(dispatch)
+  );
+}
+
+const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));
+
+const fakeITFRequest = async (url, options, success) => {
+  await delay(1000);
+  const response = new Response();
+  response.data = existingITFData;
+  return success(response);
+};
+
+// TODO: remove this mock and its helpers once user testing is complete
```

Comment:
Ahh, that's why it's still in here. Makes sense. Thanks for the clarification! :raised_hands: